File: src/main/java/com/dasoops/common/entity/enums/database/DbBooleanEnum.java
package com.dasoops.common.entity.enums.database;

import com.dasoops.common.exception.CustomException;
import com.dasoops.common.exception.DbColumnExceptionEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author DasoopsNicole@Gmail.com
 * @version 1.0.0
 * @title DbBooleanConvertTinyintEnum
 * @classPath com.dasoops.common.entity.enums.base.DbBooleanConvertTinyintEnum
 * @date 2023/01/20
 * @description db布尔值枚举
 * @see Enum
 * @see IDbColumnEnum
 */
@AllArgsConstructor
public enum DbBooleanEnum implements IDbColumnEnum {

    /**
     * 真
     */
    TRUE(1),
    /**
     * 假
     */
    FALSE(0),
    ;

    @Getter
    final int dbValue;

    public static DbBooleanEnum getBy(Integer dbValue) {
        return switch (dbValue) {
            case 1 -> TRUE;
            case 0 -> FALSE;
            default -> throw DbColumnExceptionEnum.UNDEFINEND_VALUE.getException();
        };
    }

    public static DbBooleanEnum getBy(Boolean booleanValue) {
        return booleanValue ? TRUE : FALSE;
    }

}

File: src/main/java/com/dasoops/common/entity/enums/database/DbBooleanEnum.kt
package com.dasoops.common.entity.enums.database

/**
 * @author DasoopsNicole@Gmail.com
 * @version 1.0.0
 * @title DbBooleanEnum
 * @classPath com.dasoops.common.entity.enums.database.DbBooleanEnum
 * @date 2023/02/23
 * @description 数据库boolean枚举
 * @see Enum
 *
 * @see IDbColumnEnum
 */
enum class DbBooleanEnum(
    override val dbValue: Int,
    val bool: Boolean
) : ApiEnum {
    //
    TRUE(1, true),
    FALSE(0, false),
    ;

    companion object {
        fun by(dbValue: Int): DbBooleanEnum {
            return if (dbValue == 1) {
                TRUE
            } else {
                FALSE
            }
        }

        fun by(booleanValue: Boolean): DbBooleanEnum {
            return if (booleanValue) {
                TRUE
            } else {
                FALSE
            }
        }
    }

    override val data: HashMap<String, String>?
        get() = null
}